### PR TITLE
Add a settingKey to  build separate project of the src/main and src/test

### PR DESCRIPTION
**Pull request by chank**
Add a testProjectDirectory settingKey which can build separate project of the src/main and src/test. The purpose to add this key is that, if we has a macros project, the official sbteclipse put the test code and the main code in the same project build, which is a wrong behavior.

### CODE REVIEW

File: sbteclipse-core/src/main/scala/com/typesafe/sbteclipse/core/EclipsePlugin.scala [hunk 1]
```diff
@@ -43,98 +43,84 @@ trait EclipsePlugin {
...
       commandName := "eclipse",
-      commands <+= (commandName)(Eclipse.eclipseCommand)
-    )
+      commands <+= (commandName)(Eclipse.eclipseCommand))
   }
 
   object EclipseKeys {
     import EclipseOpts._
 
+    val testProjectDirectory: SettingKey[Option[File]] = SettingKey(
```
Comment (Atry) [inline sbteclipse-core/src/main/scala/com/typesafe/sbteclipse/core/EclipsePlugin.scala]: I used settings like this
```
EclipseKeys.testProjectDirectory := Some(baseDirectory.value / s".${name.value}.test_project")
```

**Comment by benmccann** [comment]
@chank just a reminder that there's a merge conflict for this PR

**Comment by dotta** [comment]
@dragos I'm wondering if this is still needed now given http://scala-ide.org/blog/release-notes-4.1-RC1.html#new-schema-of-compiling-dependent-projects

**Comment by dragos** [comment]
nope, not needed anymore.

**Comment by dotta** [comment]
@benmccann could you close this PR

**Pull request closed without merging**